### DISCUSSION TITLE
FEType Error with Adaptivity

### DISCUSSION THREAD

**Discussion by henryg9273** [idaholab/moose#16825] [Q&A Meshing] [answered]
Dear all,

I am trying to solve fluid-chemistry-thermal diffusion PDEs with several blocks in the Navier-Stokes module:
![捕获](https://user-images.githubusercontent.com/53314562/106095192-f5631780-60f8-11eb-9392-9099495704da.PNG)

With the adaptivity block, an error message " ERROR: Bad FEType.family= 41  " always comes after the first converged step. Without the adaptivity block, the calculation can continue, but it is hard to converge. Below is the adaptivity block:

[Adaptivity]
  marker = errorfrac  #this line when commented, switches off adaptivity
  max_h_level = 3
  #steps = 2 #this line gets ignored in a transient run
  [./Indicators]
    [./error]
      type = GradientJumpIndicator
      variable = Cure
      outputs = none
    [../]
        [./error2]
      type = GradientJumpIndicator
      variable = Temperature
      outputs = none
    [../]
    
  [../]
  [./Markers]
    [./errorfrac]
      type = ErrorFractionMarker
      refine = 0.65
      coarsen = 0.2
      indicator = error
      outputs = none
    [../]
    
  [../]
[]

I'm not sure about whether the types of marker and indicator are supported in the Navier-Stokes package or if it is the cause of the error.

Can I get some suggestions on this? The full script is attached below.

Thank you in advance!
Yuan

[GlobalParams]
  supg = true
  pspg = true
  alpha = 1e-1
[]

[Mesh]
  file = FluidNew.inp
  uniform_refine = 0
  construct_side_list_from_node_list = true
  dim = 2
[]

[AuxVariables]
  [vel_x]
    order = FIRST
  []
  [vel_y]
    order = FIRST
  []
[]

[AuxKernels]
  [vel_x]
    type = VectorVariableComponentAux
    variable = vel_x
    vector_variable = velocity
    component = 'x'
  []
  [vel_y]
    type = VectorVariableComponentAux
    variable = vel_y
    vector_variable = velocity
    component = 'y'
  []
[]

[Variables]

  [./velocity]
    order = FIRST
    family = LAGRANGE_VEC
  [../]

  [./Temperature]
    order = FIRST
    family = LAGRANGE
    initial_condition = '20'
  [../]

  [./p]
  [../]
  
  [./Cure]
    order = FIRST
    family = LAGRANGE
    initial_condition = '0.3'

  [../]

[]

[Kernels]
  [./mass]
    type = INSADMass
    variable = p
  [../]
 
  [./momentum_time]
    type = INSADMomentumTimeDerivative
    variable = velocity
  [../]
  
  [./momentum_convection]
    type = INSADMomentumAdvection
    variable = velocity
  [../]

  [./momentum_viscous]
    type = INSADMomentumViscous
    variable = velocity
  [../]

  [./momentum_pressure]
    type = INSADMomentumPressure
    variable = velocity
    p = p
    integrate_p_by_parts = true
  [../]

  [./temperature_advection]
   type = INSADTemperatureAdvection
   variable = Temperature
   velocity = velocity
 [../]

 [./temperature_time]
   type = ADHeatConductionTimeDerivative
   variable = Temperature
   specific_heat = 'cp'
   density_name = 'rho'
 [../]

 [./temperature_conduction]
   type = ADHeatConduction
   variable = Temperature
   thermal_conductivity = 'k'
 [../]

  [./coupledcurederv]
    type = CoupledCureTimeDerivative
    variable = Temperature
    v = Cure
  
  [../]

  

  
  [./curederv]
  type=TimeDerivative
  variable = Cure
  lumping=false

  [../]  
 
 

 [./cureformula]
  type=DCPDnonDgeneralPT
  variable = Cure
  v = Temperature #this is the coupled variable
  Ttrig = '1'
  Tintl = '0'
  #_E = '82179.5'
  #_n = '2.2275'
  #_Kcat = '0'
  _E= '110750'
  _n= '1.7215'
  _m= '0.77'
  _cd= '14.4778'
  _ad= '0.405'
  [../]

  [./cureadvection]
  type=CureAdvection
  variable = Cure
  v = velocity
  [../]

[]

    

[BCs]
  [./no_slip]
    type = VectorFunctionDirichletBC
    variable = velocity
    boundary = Top
  [../]

  [./no_slip2]
    type = VectorFunctionDirichletBC
    variable = velocity
    boundary = Bottom
  [../]

  [./lid]
    type = VectorFunctionDirichletBC
    variable = velocity
    boundary = Left
    function_x = 'lid_function'
  [../]

  [./outlet_p]
    type = DirichletBC
    variable = p
    boundary = Right
    value = 0.0
  [../]

  [./Initiation]
    type = DirichletBC
    #type = NeumannBC
   variable = Temperature
    boundary = Left
    value = '200'    
  [../]
[]

#length is 5*30mm, Rm is 5mm
[Materials]
  [./DCPD]
block = 0
    type = GenericConstantMaterial

        prop_names = 'specific_heat Hr density TConductivity A mu rho cp k'
    prop_values = '1600000 350000000 9.8e-7 0.152 8.55e15 15000000 9.8e-7 1600000 0.152'  # polymer attributes

  [../]
  [ins_mat]
    type = INSADMaterial
    velocity = velocity
    pressure = p
    transient_term = true
    integrate_p_by_parts = true
  []

[]

[Functions]
  [./lid_function]
    # We pick a function that is exactly represented in the velocity
    # space so that the Dirichlet conditions are the same regardless
    # of the mesh spacing.
    type = ParsedFunction
    value = '5e-5*y*(4-y)'
  [../]
[]

[Preconditioning]
  [./SMP]
    type = SMP
    full = true
    solve_type = 'NEWTON'
  [../]
[]

[Executioner]
  type = Transient
  num_steps = 10000

  nl_rel_tol = 1e-5

  end_time=200

  nl_max_its=10
  l_max_its=15
  
  [./TimeStepper]
    type = ConstantDT    
    dt = 0.01
    # dt = 3.6e14 
  [../]
  
  [./TimeIntegrator]
   type = ImplicitEuler
[../]
  
  #Preconditioned JFNK (default)
  #solve_type = 'PJFNK'
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
  #petsc_options_iname = '-pc_type'
  #petsc_options_value = 'gamg'
  
[]

[Adaptivity]
  marker = errorfrac  #this line when commented, switches off adaptivity
  max_h_level = 3
  #steps = 2 #this line gets ignored in a transient run
  [./Indicators]
    [./error]
      type = GradientJumpIndicator
      variable = Cure
      outputs = none
    [../]
        [./error2]
      type = GradientJumpIndicator
      variable = Temperature
      outputs = none
    [../]
    
  [../]
  [./Markers]
    [./errorfrac]
      type = ErrorFractionMarker
      refine = 0.65
      coarsen = 0.2
      indicator = error
      outputs = none
    [../]
    
  [../]
[]

[Controls]
  [./bcs]
    type = TimePeriod
    disable_objects = 'BCs::Initiation'
    start_time = '1.0' #'1718880'  '1317808'   
    execute_on = 'initial timestep_begin'
  [../]   

[]

[Outputs] 
  execute_on = 'timestep_end' # Limit the output to timestep end (removes initial condition)
  [./console]
    type = Console
    perf_log = true          # enable performance logging    
  [../]
  [./exodus]
    type = Exodus
    execute_on = 'initial timestep_end' # output the initial condition for the file
    output_material_properties = true
    show_material_properties = 'specific_heat thermal_conductivity'
    file_base = ./Fluid/DCPD # set the file base (the extension is automatically applied) 
    interval = 10         # only output every 10 step
  [../]  
[]

**Comment by aeslaughter** [comment, marked as answer]
Can you provide the complete error message please? @lindsayad This is using Navier Stokes, not sure if it is related.

**Reply by henryg9273** [reply]
Thanks for the reply, here is the complete error message:


ERROR: Bad FEType.family= 41
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
0x00007efc6cfd94b9 in waitpid () from /lib64/libc.so.6
To enable execution of this file add
	add-auto-load-safe-path /software/gcc-7.2.0/lib64/libstdc++.so.6.0.24-gdb.py
line to your configuration file "/home/ygao73/.gdbinit".
To completely disable this security protection add
	set auto-load safe-path /
line to your configuration file "/home/ygao73/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
	info "(gdb)Auto-loading safe path"
#0  0x00007efc6cfd94b9 in waitpid () from /lib64/libc.so.6
#1  0x00007efc6cf56f62 in do_system () from /lib64/libc.so.6
#2  0x00007efc6cf57311 in system () from /lib64/libc.so.6
#3  0x00007efc71d1d454 in libMesh::print_trace(std::ostream&) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#4  0x00007efc71d19174 in libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#5  0x00007efc71e092ae in libMesh::FEGenericBase<double>::build(unsigned int, libMesh::FEType const&) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#6  0x00007efc723b6fcd in libMesh::FEGenericBase<double>* libMesh::FEMContext::cached_fe<double>(unsigned int, libMesh::FEType) const () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#7  0x00007efc723c261b in libMesh::FEGenericBase<double>* libMesh::FEMContext::build_new_fe<double>(libMesh::FEGenericBase<double> const*, libMesh::Point const&, double) const () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#8  0x00007efc723c27df in void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#9  0x00007efc72439918 in libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>::eval_at_node(libMesh::FEMContext const&, unsigned int, unsigned int, libMesh::Node const&, bool, double) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#10 0x00007efc7243dde7 in libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, libMesh::VectorValue<double>&, double) const)>, double, libMesh::VectorSetAction<double> >::ProjectVertices::operator()(libMesh::StoredRange<__gnu_cxx::__normal_iterator<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > const*, std::vector<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > >, std::allocator<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > > > >, std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > > const&) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#11 0x00007efc7243fb0b in void libMesh::Threads::parallel_reduce<libMesh::StoredRange<__gnu_cxx::__normal_iterator<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > const*, std::vector<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > >, std::allocator<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > > > >, std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > >, libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, libMesh::VectorValue<double>&, double) const)>, double, libMesh::VectorSetAction<double> >::ProjectVertices>(libMesh::StoredRange<__gnu_cxx::__normal_iterator<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > const*, std::vector<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > >, std::allocator<std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > > > >, std::pair<libMesh::Node const*, std::tuple<libMesh::Elem const*, unsigned short, std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > > > > const&, libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, libMesh::VectorValue<double>&, double) const)>, double, libMesh::VectorSetAction<double> >::ProjectVertices&) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#12 0x00007efc72454b50 in libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, libMesh::VectorValue<double>&, double) const)>, double, libMesh::VectorSetAction<double> >::project(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&) () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#13 0x00007efc72415372 in libMesh::System::project_vector(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&, int) const () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#14 0x00007efc72415f8f in libMesh::System::project_vector(libMesh::NumericVector<double>&, int) const () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#15 0x00007efc723efe03 in libMesh::System::restrict_vectors() () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#16 0x00007efc723ad589 in libMesh::EquationSystems::reinit_solutions() () from /home/ygao73/projects/moose/scripts/../libmesh/installed/lib/libmesh_opt.so.0
#17 0x00007efc73f507d9 in FEProblemBase::meshChangedHelper(bool) () from /home/ygao73/projects/moose/framework/libmoose-opt.so.0
#18 0x00007efc73f5123d in FEProblemBase::adaptMesh() () from /home/ygao73/projects/moose/framework/libmoose-opt.so.0
#19 0x00007efc73a5a603 in Transient::incrementStepOrReject() () from /home/ygao73/projects/moose/framework/libmoose-opt.so.0
#20 0x00007efc73a5a379 in Transient::execute() () from /home/ygao73/projects/moose/framework/libmoose-opt.so.0
#21 0x00007efc7429cdf9 in MooseApp::executeExecutioner() () from /home/ygao73/projects/moose/framework/libmoose-opt.so.0
#22 0x00007efc7429b638 in MooseApp::run() () from /home/ygao73/projects/moose/framework/libmoose-opt.so.0
#23 0x0000000000402467 in main ()
[Inferior 1 (process 42867) detached]
[0] /home/ygao73/projects/moose/scripts/../libmesh/src/fe/fe_base.C, line 321, compiled Jan  5 2021 at 18:07:20
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

**Reply by lindsayad** [reply]
This looks like *potentially* a bug with adaptivity with vector variables with subdomain restriction. @henryg9273 if you could pare your input file down to only objects in the MOOSE repository and still reproduce this error message, then I can try to debug this for you.

**Reply by henryg9273** [reply]
Thank you, Alex, for the quick reply.

I was using a user-built app that contains N-S package and some user-built kernel. I'm not sure if it is equivalent to what you suggested.

**Reply by lindsayad** [reply]
Yes I assume the 'cure' objects are you own application's objects. You can either take those out and see whether you can still reproduce the error, or you can grant me access to your application and then I can build your application and run your input file unmodified.

**Reply by henryg9273** [reply]
Thanks so much, Alex. I have invited you to the repository.

**Reply by lindsayad** [reply]
It doesn't appear that you included the mesh in your repository. I cannot run your input file without it

**Reply by henryg9273** [reply]
Sorry, I have just uploaded.

**Reply by lindsayad** [reply]
Ok you must be using a really old version of MOOSE because I get this error:
```
*** ERROR ***
The requested AD material property mu is declared as a regular material property. Either retrieve it as a regular material property with getMaterialProperty or declare it as an AD property wtih declareADProperty
```
This is an error application developers/analysts should have addressed almost a year ago. If I run into more of this, I won't be able to debug your issue at all

**Reply by lindsayad** [reply]
Ok after updating your input file, I do not get the error you reported. It appears that you were using quite an old version of MOOSE, and it's possible that the error you saw has since been fixed. Please update your MOOSE and then make sure you update libmesh as well.
